### PR TITLE
Hw1 project4

### PR DESCRIPTION
**Pull request by OuChiaYun**
不確定提交檔案的要求是不是這樣子，沒有放.ko
盡量寫了點文件

### CODE REVIEW

File: project_4/part1/Makefile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+obj-m:= color.o
+
+kernel_DIR:= /usr/src/linux-headers-5.4.0-148-generic/
```
Comment (x06lan) [inline project_4/part1/Makefile]: avoid use hardcode path
consider use $(shell uname -r)

File: .vs/POSIXat/v15/.suo
Comment (x06lan) [inline .vs/POSIXat/v15/.suo:1]: remove this

File: .vs/VSWorkspaceState.json
Comment (x06lan) [inline .vs/VSWorkspaceState.json:1]: remove this

File: project_4/part2/Makefile [hunk 2]
```diff
@@ -0,0 +1,12 @@
+obj-m:= collatz.o
+
+kernel_DIR:= /usr/src/linux-headers-5.4.0-148-generic/
```
Comment (x06lan) [inline project_4/part2/Makefile]: same

**Review by nonorze-VOXX — COMMENTED** [review]
if remove .vs can it work?
y -> add .vs to gitignore

File: project_4/part1/Makefile (hunk 1 above)
Comment (nonorze-VOXX) [inline project_4/part1/Makefile]: maybe can ref #2

**Review by x06lan — CHANGES_REQUESTED** [review]

File: project_4/part1/Makefile (hunk 1 above)
Comment (OuChiaYun) [reply project_4/part1/Makefile]: done

File: .vs/POSIXat/v15/.suo
Comment (OuChiaYun) [reply .vs/POSIXat/v15/.suo:1]: done

**Comment by OuChiaYun** [comment]
> if remove .vs can it work? y -> add .vs to gitignore

done, add .gitignore and add rm .vs

File: project_4/part2/Makefile (hunk 2 above)
Comment (OuChiaYun) [reply project_4/part2/Makefile]: done

File: .vs/VSWorkspaceState.json
Comment (OuChiaYun) [reply .vs/VSWorkspaceState.json:1]: done, use .gitignore and rm .vs

**Review by x06lan — APPROVED** [review]

**Pull request merged**